crates/evm/src/env.rs

```diff
@@ -1,36 +1,26 @@
 use revm::primitives::{BlockEnv, CfgEnvWithHandlerCfg};
+use revm_primitives::{CfgEnv, SpecId};
 
 /// Container type that holds both the configuration and block environment for EVM execution.
-#[derive(Debug, Clone)]
-pub struct EvmEnv {
+#[derive(Debug, Clone, Default)]
+pub struct EvmEnv<Spec = SpecId> {
     /// The configuration environment with handler settings
-    pub cfg_env_with_handler_cfg: CfgEnvWithHandlerCfg,
+    pub cfg_env: CfgEnv,
     /// The block environment containing block-specific data
     pub block_env: BlockEnv,
+    /// The spec id of the chain
+    pub spec: Spec,
```

Comment:
I feel like this doc isn't quite accurate enough because the spec is the evm spec which kinda is more or less the same as the chain's hardfork

so I'd like to clarify here that this represents the evm spec of the chain and that this is more or less equivalent to the chain's hardfork variants